Tests/Indicators/McClellanSummationIndexTests.cs

```diff
@@ -0,0 +1,153 @@
+                indicator.Update(new TradeBar() { Symbol = Symbols.MSFT, Close = i, Volume = 1, Time = reference.AddMinutes(i) });
+                indicator.Update(new TradeBar() { Symbol = Symbols.GOOG, Close = i, Volume = 1, Time = reference.AddMinutes(i) });
+            }
+
+            Assert.AreEqual(0m, indicator.Current.Value);
+            Assert.AreEqual(indicator.WarmUpPeriod * 3, indicator.Samples);
+            Assert.IsTrue(indicator.IsReady);
+        }
+
+        [Test]
+        public override void ResetsProperly()
+        {
+            var indicator = (McClellanSummationIndex)CreateIndicator();
+            var reference = DateTime.Today;
+
+            for (int i = 1; i <= indicator.WarmUpPeriod; i++)
+            {
+                indicator.Update(new TradeBar() { Symbol = Symbols.AAPL, Close = i, Volume = 1, Time = reference.AddMinutes(i) });
+                indicator.Update(new TradeBar() { Symbol = Symbols.MSFT, Close = i, Volume = 1, Time = reference.AddMinutes(i) });
+                indicator.Update(new TradeBar() { Symbol = Symbols.GOOG, Close = i, Volume = 1, Time = reference.AddMinutes(i) });
+            }
+
+            Assert.IsTrue(indicator.IsReady);
+
+            indicator.Reset();
+
+            TestHelper.AssertIndicatorIsInDefaultState(indicator);
+        }
+
+        [Test]
+        public override void ComparesAgainstExternalData()
+        {
+            var indicator = new TestMcClellanSummationIndex();
+            McClellanIndicatorTestHelper.RunTestIndicator(indicator, TestFileName, TestColumnName);
+        }
+
+        [Test]
+        public override void ComparesAgainstExternalDataAfterReset()
+        {
+            var indicator = new TestMcClellanSummationIndex();
+            McClellanIndicatorTestHelper.RunTestIndicator(indicator, TestFileName, TestColumnName);
+            indicator.Reset();
+            McClellanIndicatorTestHelper.RunTestIndicator(indicator, TestFileName, TestColumnName);
+        }
+
+        protected override string TestFileName => "mcclellan_data.csv";
+
+        protected override string TestColumnName => "MSI";
+    }
+
+    public class TestMcClellanSummationIndex : McClellanSummationIndex, ITestMcClellanOscillator
+    {
+        private Dictionary<Symbol, decimal> _symbols = new();
+        private int _dateCount = 1;
+
+        public TestMcClellanSummationIndex() : base()
+        {
+            // Maximum A/D difference from the test set is 2527
+            for (int i = 1; i <= 2530; i++)
+            {
+                var symbol = Symbol.Create($"TestSymbol{i}", SecurityType.Equity, Market.USA);
+                _symbols.Add(symbol, 0m);
+                Add(symbol);
+            }
+
+            // Set to the first EMA values to account for past A/D Difference values that we don't have access
+            Reset();
+            _mcClellanSummationIndex.Time = new DateTime(2022, 6, 30);
+            _mcClellanSummationIndex.Value = -629.68m;
```

Comment:
Please use `-606.25m` to fix the unit tests. We need to use some reverse engineering here.